### PR TITLE
Check errors and delete pod at end of portforward test

### LINKED ISSUE #21332: Pod failed to start and cause parsing error
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/portforward.go:202
Feb 13 10:54:35.120: Failed to parse kubectl port-forward output: F0213 10:54:35.119866   25647 portforward.go:115] Unable to execute command because pod is not running. Current status=Pending
```

https://console.developers.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/20351/kubernetes-pull-build-test-e2e-gce/28440/artifacts/

Two issues here: Pod failed to start, and test failed to parse the output.

### PR DESCRIPTION
**Pull request by davidopp**
Might fix #21332

cc/ @ncdc

### CODE REVIEW

**Comment by ncdc** [comment]
Will review tomorrow

File: test/e2e/portforward.go [hunk 1]
```diff
@@ -153,13 +161,25 @@ var _ = Describe("Port forwarding", func() {
 			logOutput := runKubectlWithTimeout(wait.ForeverTestTimeout, "logs", fmt.Sprintf("--namespace=%v", framework.Namespace.Name), "-f", podName)
 			verifyLogMessage(logOutput, "Accepted client connection")
 			verifyLogMessage(logOutput, "Expected to read 3 bytes from client, but got 0 instead")
+			poderr = framework.Client.Pods(framework.Namespace.Name).Delete(pod.Name,
```
Comment (ncdc) [inline test/e2e/portforward.go]: The pod deletions aren't strictly required; `framework` has an `AfterEach` that deletes the namespace.

File: test/e2e/portforward.go [hunk 2]
```diff
@@ -129,13 +129,21 @@ func runKubectlWithTimeout(timeout time.Duration, args ...string) string {
...
+	var zero int64
 
 	Describe("With a server that expects a client request", func() {
 		It("should support a client that connects, sends no data, and disconnects [Conformance]", func() {
 			By("creating the target pod")
+			var poderr error
 			pod := pfPod("abc", "1", "1", "1")
-			framework.Client.Pods(framework.Namespace.Name).Create(pod)
-			framework.WaitForPodRunning(pod.Name)
+			_, poderr = framework.Client.Pods(framework.Namespace.Name).Create(pod)
```
Comment (ncdc) [inline test/e2e/portforward.go]: I would probably just do `if err := ...; err != nil {`

**Comment by ncdc** [comment]
@davidopp I'd be happy to roll this into #21602 if you wanted me to

**Comment by davidopp** [comment]
@ncdc OK, I'll close this PR, please incorporate into #21602.

**Pull request closed without merging**